extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -285,12 +290,51 @@ export class ResourceTypePickerDialog extends DialogBase {
 		this.resourceTypeService.startDeployment(this.getCurrentProvider());
 	}
 
-	private installTools(): void {
-		this._dialogObject.message = {
-			level: azdata.window.MessageLevel.Information,
-			text: localize('deploymentDialog.InstallingTools', "Required tools are being installed now.")
-		};
+	private UpdateToolsTableData(tool: ITool) {
+		this._toolsTable.data = this._toolsTable.data.map(rowData => {
+			if (rowData[0] === tool.displayName) {
+				if (tool.status === ToolStatus.NotInstalled || tool.status === ToolStatus.Error) {
```

Comment:
Not a huge deal, but would be faster if you just cached the table in a map-like structure such that you could do a direct lookup instead of having to loop through them all

Reply:
Agreed. But this issue pre-exists in more places than one. I will come back and fix in a separate PR.